packages/api/lib/granule-remove-from-cmr.js

```diff
@@ -41,17 +41,19 @@ const unpublishGranule = async (knexOrTransaction, granule) => {
   const granulePgModel = new GranulePgModel();
   const collectionPgModel = new CollectionPgModel();
 
+  let pgGranule;
+
   await _removeGranuleFromCmr(granule);
 
-  // If we cannot find a PG Collection for this granule,
+  // If we cannot find a PG Collection or PG Granule,
   // don't update the PG Granule, continue to update the Dynamo granule
   try {
     const collectionCumulusId = await collectionPgModel.getRecordCumulusId(
       knexOrTransaction,
       deconstructCollectionId(granule.collectionId)
     );
 
-    await granulePgModel.update(
+    [pgGranule] = await granulePgModel.update(
       knexOrTransaction,
       {
         granule_id: granule.granuleId,
```

Comment:
That's a really good point. Whatever we do for deletes should probably also be on updates/upserts. 

@markdboyd did you end up settling on an approach for validating keys on delete? I still like the idea of only using `cumulus_id` from the perspective of the `delete` method but there is a definite downside. If our code doesn't pass the cumulus_id everywhere we need it, we'll need to add queries to get the ID every time.

Reply:
I didn't decide on an approach but I was planning a follow up PR, notionally as part of 2367, to address it

Reply:
Should this perhaps be a separate ticket so that we can address it on deletes as well? Also happy to do it here but I just want to be sure we have a consistent approach.

Reply:
I had hesitated on going the "another ticket" approach for fear of the work getting lost. I might take a look at seeing if I can at least work something up for "delete" this morning, but if I'm not getting anywhere I can create a ticket

Reply:
I think so long as we've captured the concern it doesn't need to be in this PR, but we absolutely shouldn't consider Phase 1 done until we address. this.